sdk/typescript/src/module/entrypoint/entrypoint.ts

```diff
@@ -71,3 +67,35 @@ export async function entrypoint(files: string[]) {
     { LogOutput: process.stdout },
   )
 }
+
+/**
+ * Take the error thrown by the user module and stringify it so it can
+ * be returned to Dagger.
+ *
+ * If the error is an instance of Error, we stringify the message.
+ * If the error is an instance of ExecError, we stringify the message and add the
+ * extensions fields in the error object.
+ */
+function formatError(e: unknown): DaggerError {
+  if (e instanceof Error) {
+    let error = dag.error(e.message)
+
+    // If the error is an instance of GraphQLError or a inherit type of it,
+    // we can add the extensions fields in the error object.
+    if (e instanceof ExecError || e instanceof GraphQLRequestError) {
+      Object.entries(e.extensions ?? []).forEach(([key, value]) => {
+        if (value !== "" && value !== undefined && value !== null) {
+          error = error.withValue(key, JSON.stringify(value) as JSON)
+        }
+      })
+    }
+
+    return error
+  }
+
+  try {
+    return dag.error(JSON.stringify(e))
+  } catch {
+    return dag.error(String(e))
+  }
```

Comment:
What's this meant to handle? :thinking: Looks a little odd but maybe because I'm missing JS/TS knowledge.

I guess this is because non-`Error` values can be thrown? Should we even turn those into errors?

Reply:
Yeah it's possible in case of a "panic" in the code, for example if the user tries to access an undefined value in its module, in these case, I'm not sure the exception will be in an `Error` object so it's an extra safety measure to correctly return the error to the user instead of crashing and display that `tsx xxx` error message.